argilla-server/src/argilla_server/telemetry.py

```diff
@@ -17,114 +17,195 @@
+            "count_questions": len(dataset.questions),
+            "count_vector_settings": len(dataset.vectors_settings),
+            "count_metadata_properties": len(dataset.metadata_properties),
+            "allow_extra_metadata": dataset.allow_extra_metadata,
+            "guidelines": True if dataset.guidelines else False,
+        }
 
+    @staticmethod
+    def _process_dataset_setting_settings(setting: Any):
+        return {}
 
-def track_user_created(user: User, is_oauth: bool = False):
-    _CLIENT.track_data(
-        action="UserCreated",
-        data={
+    @staticmethod
+    def _process_user_model(user: User):
+        return {
             "user_id": str(user.id),
             "role": user.role,
             "is_default_user": user.username == DEFAULT_USERNAME,
-            "is_oauth": is_oauth,
-        },
-    )
+            "user_hash": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=user.username)),
+        }
+
+    def track_data(self, topic: str, user_agent: dict, include_system_info: bool = True, count: int = 1):
+        library_name = "argilla"
+        topic = f"{library_name}/{topic}"
+
+        if include_system_info:
+            user_agent.update(self._system_info)
+        if count is not None:
+            user_agent["count"] = count
+
+        send_telemetry(topic=topic, library_name=library_name, library_version=__version__, user_agent=user_agent)
+
+    async def track_user_login(self, request: Request, user: User):
+        topic = "user/login"
+        user_agent = self._process_user_model(user=user)
+        user_agent.update(**self._process_request_info(request))
+        self.track_data(topic=topic, user_agent=user_agent)
+
+    async def track_crud_user(
+        self,
+        action: str,
+        user: Union[User, None] = None,
+        is_oauth: Union[bool, None] = None,
+        count: Union[int, None] = None,
+    ):
+        topic = f"user/{action}"
+        if user:
+            user_agent = self._process_user_model(user=user)
+        if is_oauth is not None:
+            user_agent["is_oauth"] = is_oauth
+        self.track_data(topic=topic, user_agent=user_agent, count=count)
+
+    async def track_crud_workspace(
+        self, action: str, workspace: Union[Workspace, None] = None, count: Union[int, None] = None
+    ):
+        topic: str = f"workspace/{action}"
+        if workspace:
+            user_agent = self._process_workspace_model(workspace=workspace)
+        self.track_data(topic=topic, user_agent=user_agent, count=count)
+
+    async def track_crud_dataset(
+        self, action: str, dataset: Union[Dataset, None] = None, count: Union[int, None] = None
+    ):
+        topic = f"dataset/{action}"
+        if dataset:
+            user_agent = self._process_dataset_model(dataset=dataset)
+            user_agent.update(self._process_dataset_settings(dataset=dataset))
+        self.track_data(topic=topic, user_agent=user_agent, count=count)
+
+        return None
```

Comment:
Is this intended?

Reply:
Yes, I need to review and update the processing for these settings methods.